zookeeper-server/src/main/java/org/apache/zookeeper/server/ZKDatabase.java

```diff
@@ -120,6 +124,23 @@ public ZKDatabase(FileTxnSnapLog snapLog) {
             snapshotSizeFactor = DEFAULT_SNAPSHOT_SIZE_FACTOR;
         }
         LOG.info("{} = {}", SNAPSHOT_SIZE_FACTOR, snapshotSizeFactor);
+
+        try {
+            commitLogCount = Integer.parseInt(
+                    System.getProperty(COMMIT_LOG_COUNT,
+                            Integer.toString(DEFAULT_COMMIT_LOG_COUNT)));
+            if (commitLogCount < DEFAULT_COMMIT_LOG_COUNT) {
+                commitLogCount = DEFAULT_COMMIT_LOG_COUNT;
+                LOG.warn("The configured commitLogCount {} is less than the recommended {}"
```

Comment:
I think this would have been added to the documentation. The default value of 500 is not just a default value, but the minimum and the recommended one. What's the risk of setting this to less than 500 btw?

Reply:
I can create another PR to update the docs and make sure I say the value is the recommended minimum and that it defaults to 500. As for the risk of using a lower value, I am guessing @enixon wanted to make sure we have some limit that reflects a well known best practice. 

Should I change that? I personally like that we have some safety enforcement around the config values.

Reply:
Changing the docs only should be fine I think.

Reply:
Created this PR https://github.com/apache/zookeeper/pull/1089 to address the comment from above. Let me know if you had something else on your mind so I can quickly update otherwise that's it.